python/kserve/kserve/protocol/infer_type.py

```diff
@@ -562,7 +647,14 @@ def as_numpy(self) -> numpy.ndarray:
         if dtype is None:
             raise InvalidInput("invalid datatype in the input")
         if self._raw_data is not None:
-            np_array = np.frombuffer(self._raw_data, dtype=dtype)
+            if self.datatype == "BYTES":
+                # String results contain a 4-byte string length
+                # followed by the actual string characters. Hence,
+                # need to decode the raw bytes to convert into
+                # array elements.
+                np_array = deserialize_bytes_tensor(self._raw_data)
```

Comment:
This is only for the REST case right? for gRPC we can just call np.frombuffer

Reply:
Triton client uses serialization for both http and grpc